### PR TITLE
PR to address incorrect package versions in DataProtection Configuration document (Issue #19770)

### LINKED ISSUE #19770: DataProtection Configuration references the wrong libraries
# Issue description

The configuration document for DataProtection references the incorrect packages, which include issues requiring awkward workarounds.

I'll create a PR to change 

Microsoft.AspNetCore.DataProtection.AzureStorage
Microsoft.AspNetCore.DataProtection.AzureKeyVault

to 

Azure.Extensions.AspNetCore.DataProtection.Blobs
Azure.Extensions.AspNetCore.DataProtection.Keys

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

- [x ] .NET Core - 3.1 
- [ ] .NET Framework
- [ ] .NET Standard

### PR DESCRIPTION
**Pull request by markmcgookin**
Fixes #19770

Update to fix package references and minor configuration changes when moving to the new libraries.

Left the workaround in place for the issue with the old library incase people are googling the error and come across the document (included a comment suggesting them to upgrade)

### CODE REVIEW

**Comment by guardrex** [comment]
@Rick-Anderson ... This isn't the only spot for the updates. I don't see any sample apps that need work, but there are perhaps four more topics that need work.

**Comment by Rick-Anderson** [comment]
@pakrym please review or recommend a reviewer

File: aspnetcore/security/data-protection/configuration/overview.md [hunk 1]
```diff
@@ -40,9 +40,12 @@ To store keys in [Azure Key Vault](https://azure.microsoft.com/services/key-vaul
 ```csharp
 public void ConfigureServices(IServiceCollection services)
 {
+    //Uri to the key in azure keyvault that is being used to protect the keys in blob storage
+    var keyUri = new Uri($"https://{vaultName}.vault.azure.net/keys/{keyname}");
+    
     services.AddDataProtection()
-        .PersistKeysToAzureBlobStorage(new Uri("<blobUriWithSasToken>"))
-        .ProtectKeysWithAzureKeyVault("<keyIdentifier>", "<clientId>", "<clientSecret>");
+                    .PersistKeysToAzureBlobStorage("<storageAccountConnectionString>", "<storageContainerName>", "<keyStorageBlobName>")
```
Comment (pakrym) [inline aspnetcore/security/data-protection/configuration/overview.md:47]: The overload that takes blob URI still exists, I would recommend keeping using it along with the `DefaultAzureCredential` from `Azure.Identity`

Comment (markmcgookin) [reply aspnetcore/security/data-protection/configuration/overview.md]: Ok, i've rolled that back to the original overload. I must have missed it when setting up my project and moved over to the other one.

**Comment by markmcgookin** [comment]
I've updated the PR at @pakrym 's request to have the original setup arguments, I changed mine when migrating to the new libraries, but apparently the original ones in the documentation still work.

**Review by pakrym — APPROVED** [review]

**Review by guardrex — APPROVED** [review]
Thanks @markmcgookin! :rocket:

**Pull request merged**